raiden/api/rest.py

```diff
@@ -431,11 +432,18 @@ def patch_channel(self, channel_address, balance=None, state=None):
                     'Attempted to settle a channel at its {} state'.format(current_state),
                     httplib.CONFLICT,
                 )
-            raiden_service_result = self.raiden_api.settle(
-                channel.token_address,
-                channel.partner_address
-            )
-            result = self.channel_schema.dump(channel_to_api_dict(raiden_service_result))
+            try:
+                raiden_service_result = self.raiden_api.settle(
+                    channel.token_address,
+                    channel.partner_address
+                )
+            except InvalidState:
+                result = make_response(
+                    'Settlement period is not yet over',
+                    httplib.BAD_REQUEST,
```

Comment:
Hmm, I thought it was more of a convention.. anyway, commiting 409 and merging request. Thanks.